### PR TITLE
Swift code style

### PR DESCRIPTION
**Pull request by dnerdy**
A draft code style document for Swift.

Status: **Opened for visibility**
 
Reviewers: @jeremywiebe @jvaill @noahadams

### CODE REVIEW

File: swift/README.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+You have a couple options:
+
+- Return a tuple (similar to Go)
+- Use a result/error enumeration type
+
+We typically use an enumeration type:
+
+```
+public enum ParseResult {
+    case Result(NSObject)
```
Comment (dnerdy) [inline swift/README.md]: Shouldn't use NSObject here as an example.

File: swift/README.md [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+}
+```
+
+`if let` variables
+==================
+
+Use shadowing:
+
+```
+let message: String?
```
Comment (dnerdy) [inline swift/README.md]: `let message: String? = "hello"`

**Comment by dnerdy** [comment]
Edits look great. Thanks @jeremywiebe!

**Comment by lizcross** [comment]
Right now there aren't too many iOS specific items. The main one I see if for the preferred layout type. We could think about making an iOS specific file inside the swift folder if more iOS specific items are added.

**Comment by jansepar** [comment]
I think this is a great start - anyone else we need to address before merging?

**Comment by jansepar** [comment]
:+1:

**Pull request merged**